plugins/modules/site_intent.py

```diff
@@ -774,10 +783,47 @@ def get_diff_deleted(self, config):
         site_exists = self.have.get("site_exists")
 
         if site_exists:
+            # Check here if the site have the childs then fetch it using get membership API and then sort it
+            # in reverse order and start deleting from bottom to top
+            site_id = self.have.get("site_id")
+            mem_response = self.dnac._exec(
+                family="sites",
+                function="get_membership",
+                params={"site_id": site_id},
+            )
+            site_response = mem_response.get("site").get("response")
+
+            if len(site_response) > 0:
+                # Sorting the response in reverse order based on hierarchy levels
+                sorted_site_resp = sorted(site_response, key=lambda x: x.get("groupHierarchy"), reverse=True)
```

Comment:
Looks like line number 823 to 844 are redundant code. Can we make it as a function "delete_single_site" and optimize the code? You can set and return the self with proper values

Sample code as follows:

```
        for item in sorted_site_resp:
            self.delete_single_site(item['id'], item['name'])

        self.delete_single_site(site_id, self.want.get("site_name"))

# New API
def delete_single_site(self, site_id, site_name):
    response = self.dnac._exec(
        family="sites",
        function="delete_site",
        params={"site_id": site_id},
    )

    if response and isinstance(response, dict):
        executionid = response.get("executionId")
        while True:
            execution_details = self.get_execution_details(executionid)
            if execution_details.get("status") == "SUCCESS":
                self.result['changed'] = True
                self.log("Site - {0} deleted successfully".format(site_name))
                break
            elif execution_details.get("bapiError"):
                self.module.fail_json(msg=execution_details.get("bapiError"), response=execution_details)
                break

```

Reply:
added.